sunpy/map/sources/yohkoh.py

```diff
@@ -56,7 +56,7 @@ def __init__(self, data, header, **kwargs):
         # use simple trigonometry to calculate the distance of the center of
         # the Sun from the spacecraft.  Note that the small angle approximation
         # is used, and the solar radius stored in SXT FITS files is in arcseconds.
-        self.meta['dsun_apparent'] = constants.au
+        self.meta['dsun_apparent'] = self.meta.get('dsun_apparent', constants.au)
         if 'solar_r' in self.meta:
             self.meta['dsun_apparent'] = constants.radius/(np.deg2rad(self.meta['solar_r']/3600.0))
```

Comment:
```suggestion
        if 'dsun_apparent' not in self.meta:
            if 'solar_r' in self.meta:
                self.meta['dsun_apparent'] = constants.radius/(np.deg2rad(self.meta['solar_r']/3600.0))
            else:
                self.meta['dsun_apparent'] = (sun.earth_distance(self.date)).to(u.m).value
```
And import `sun` from `sunpy.coordinates` up top.

Reply:
Some comments:

- `dsun_apparent` is a SunPy-generated keyword, so we could choose to not trust it in a file even if it's there.
- I hate the 1 AU assumption.
- I don't know whether SXT files ever don't have `solar_r`, so maybe that code branch isn't actually used.

Reply:
I think changing/fixing the dsun_apparent keyword from au to the actual sun-earth distance is a bugfix that's orthogonal to this PR.

Reply:
I have moved this to it's own issue as I want to get this PR in before 2.1.